src/estimators/independent_set_estimator.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2020 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Estimator for sketches when using the independence assumption.
+
+Under the independence assumption, the cardinality of the union of two
+sketches representing sets A and B is equal to
+   
+  |A| + |B| - |A| * |B| / N,
+
+where N is the size of the universe from which A and B are drawn.
+"""
+
+from itertools import accumulate
+from wfa_cardinality_estimation_evaluation_framework.estimators.base import EstimatorBase
+
+
+class IndependentSetEstimator(EstimatorBase):
+  """An estimator that estimates the size of a union by assuming independence.
+
+  Given two sketches A and B representing samples drawn from a universe U,
+  the estimated size of A union B is given by:
+
+    |A union B| = |A| + |B| - |A| * |B| / |U|.
+
+  """
+
+  def __init__(self, single_sketch_estimator, universe_size):
+    """Instantiates an IndependentSetEstimator object.
+
+    Args:
+      single_sketch_estimator:  An object derived from EstimatorBase for
+        providing cardinality estimates of individual sketches.
+      universe_size: Size of the universe.
+    """
+    self.single_sketch_estimator = single_sketch_estimator
+    self.universe_size = universe_size
+
+  def __call__(self, sketch_list):
```

Comment:
sketch_list requires frequency sketch representation? Need to sync on this structure. The stratefied sketch uses per freq sketch.

Reply:
I don't think there is a conflict with StratifiedSketch here.